src/sentry/shared_integrations/response/sequence.py

```diff
@@ -1,11 +1,15 @@
+from __future__ import annotations
+
+from typing import Any, Mapping, Sequence
+
 from sentry.shared_integrations.response.base import BaseApiResponse
 
 
-class SequenceApiResponse(list, BaseApiResponse):
-    def __init__(self, data, *args, **kwargs):
+class SequenceApiResponse(list, BaseApiResponse):  # type: ignore
+    def __init__(self, data: Sequence[Any], *args: Any, **kwargs: Any) -> None:
         list.__init__(self, data)
         BaseApiResponse.__init__(self, *args, **kwargs)
 
     @property
-    def json(self):
-        return self
+    def json(self) -> Mapping[str, Any]:
```

Comment:
this subclasses `list` and at the same time is a `Mapping[str, Any]`, something smells fishy. Would change type of `json` in baseclass to `Any` because apparently you really can't assume more than that.

Alternatively you can make `BaseApiResponse` generic over a `T`, so that you can do:


```
T = TypeVar("T")

class BaseApiResponse(Generic[T]):
    json: T

class SequenceApiResponse(list, BaseApiResponse[list]):
    def json(self) -> list: ...
    
class MappingApiResponse(dict, BaseApiResponse[dict]):
    def json(self) -> dict: ...
```